engine/src/main/scala/cromwell/webservice/CromwellApiService.scala

```diff
@@ -53,14 +53,19 @@ trait CromwellApiService extends HttpService with PerRequestCreator {
       perRequest(requestContext, metadataBuilderProps, message)
   }
 
-  private def failBadRequest(t: Throwable, statusCode: StatusCode = StatusCodes.BadRequest) = respondWithMediaType(`application/json`) {
+  def handleQueryMetadataRequest(parameters: Seq[(String, String)]): Route = {
```

Comment:
this made me cry a bit. I needed to override `handleMetadataRequest` in order to not actually have requests going into Real Cromwell. The query stuff worked a bit different and I wasn't able to figure out enough spray-fu to avoid just doing this.